RockPaperScissors/RockPaperScissors/main.swift

```diff
@@ -6,5 +6,148 @@
 
 import Foundation
 
-print("Hello, World!")
+var shouldRunGame = true
 
+var validInputRange: ClosedRange<Int> {
+    return 0...RockPaperScissors.allCases.count
+}
+
+enum Script: String, CustomStringConvertible {
+    var description: String {
+        self.rawValue
+    }
+    
+    case menu = "가위(1), 바위(2), 보(3)! <종료 : 0> : "
+    case win = "이겼습니다!"
+    case lose = "졌습니다!"
+    case draw = "비겼습니다!"
+    case gameEnd = "게임 종료"
+    case dismissedError = "잘못된 입력입니다. 다시 시도해주세요."
+    case choiceOutOfRangeError = "가위바위보와 매칭될 수 있는 유효값이 아닙니다."
+}
+
+enum RockPaperScissorsGameError: Error {
+    case invalidInput
+    case choiceOutOfRange
+}
+
+enum RockPaperScissorsJudgement {
+    case win
+    case lose
+    case draw
+}
+
+enum RockPaperScissors: CaseIterable {
+    // 가위 바위 보 외의 case가 추가된다면 게임 결과를 계산해줄 matchJudgement와 matchHandPose 또한 업데이트 해야합니다.
+    case rock
+    case paper
+    case scissors
+    
+    static func getRange() -> ClosedRange<Int> {
+        return 1...Self.allCases.count
+    }
+}
+
+func manageRockPaperScissorsGame() {
+    while shouldRunGame {
+        do {
+            try startRockPaperScissorsGame()
+        } catch RockPaperScissorsGameError.invalidInput {
+            print(Script.dismissedError)
+        } catch RockPaperScissorsGameError.choiceOutOfRange {
+            print(Script.choiceOutOfRangeError)
+        } catch {
+            print(error)
+        }
+    }
```

Comment:
단순히 에러를 출력하는 동작만 수행한다면 아래와 같이 작성할 수도 있을거예요. `RockPaperScissorsError` 타입에도 `CustomStringConvertible` 프로토콜을 이용하여 에러 문구를 지정해주시면 발생한 에러에 따라 다양한 에러 문구를 보여줄 수 있겠죠?

```suggestion
        do {
            try startRockPaperScissorsGame()
        } catch {
            print(error)
        }
    }
```